com.unity.shadergraph/Documentation~/Custom-Interpolators.md

```diff
@@ -0,0 +1,65 @@
+# Custom Interpolators
+
+## Description
+
+The Custom Interpolator feature provides fine-grained control over the specific calculations Shader Graph uses to bring data from the vertex stage to the pixel stage.
+
+There are two target audiences for Custom Interpolators:
+
+* Technical Directors and Lead Technical Artists setting up environments for their teams.
+* Graphics programmers helping artists to optimize content performance.
+
+
+## Supported data types
+Custom interpolators support float, vec1, vec2, vec 3, and vec4 options.
+
+## Channel limits
+The Custom Interpolator feature supports a maximum of 32 channels. A channel is equivalent to four floats. Each float is an interpolator variable.
+Different platforms and GPUs have different interpolator variable limits. Exceeding the interpolator limitations of your target platform prevents your shaders from compiling. For detailed information about the number of interpolators supported by common interfaces, see the Unity documentation on [Shader semantics](https://docs.unity3d.com/Manual/SL-ShaderSemantics.html), and view the section **Interpolator count limits**. Test your Custom Interpolators on your target configuration to ensure that your content compiles properly.
+Technical directors can set warnings and errors to help their team members avoid creating graphs with too many channels to be compatible with their target pipeline, platform, or GPU. See **Creating channel warnings and errors** below.
+
+## How to use
+To use this feature, create a Custom Interpolator block in the Vertex context of the Master Stack and set a name and a data type. Create a vertex node to write data to that interpolator. Use the interpolator in your graph, then connect your graph to the relevant block in the Fragment context.
+These instructions include a contextual example illustrating the process of using a Custom Interpolator to fetch per-vertex data from a texture.
+To read the HLSL you use to replicate this behavior with the Built In Render Pipeline, see the Unity documentation on [Shader semantics](https://docs.unity3d.com/Manual/SL-ShaderSemantics.html) and view the section **Vertex ID: SV_VertexID**.
+
+### Creating channel warnings and errors
+
+It is not possible to limit the number of channels a user can create in a Shader Graph. However, it is possible to create alerts to let users know when they are close to or exceeding a certain number of channels.
+The **Error Threshold** lets users know that they are approaching the channel limit, and the **Warning Threshold** informs them if they have reached or surpassed that limit. The **Error Threshold** value must be between 8 and 32 channels. The **Warning Threshold** value must be higher than the **Error Threshold**, and has a minimum value of 8 channels.
```

Comment:
I followed Esme's guidance on this one

Reply:
@esmelusina is this right as-is?

Reply:
Oh, yes- that does look flipped.

"The **Warning Threshold** lets users know they are approaching the channel limit, the **Error Threshold** informs them if they have reached or surpassed that limit."

"The error threshold must be between 8 and 32" and "the warning threshold must be less than or equal to the warning threshold, and has a minimum of 8 channels".

I think I may have mixed up terms initially when talking with Val. This can be worded in a few ways though.

Reply:
Good to hear--I had thought it was odd but followed the guidance received on the doc during tech review. I'll reverse these now.